samples/sensor/accel_polling/src/main.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2022 TOKITA Hiroshi <tokita.hiroshi@fujitsu.com
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/sensor.h>
+#include <zephyr.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+static const struct device *sensors[] = {
+#if DT_NODE_EXISTS(DT_ALIAS(accel0))
+	DEVICE_DT_GET(DT_ALIAS(accel0)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel1))
+	DEVICE_DT_GET(DT_ALIAS(accel1)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel2))
+	DEVICE_DT_GET(DT_ALIAS(accel2)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel3))
+	DEVICE_DT_GET(DT_ALIAS(accel3)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel4))
+	DEVICE_DT_GET(DT_ALIAS(accel4)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel5))
+	DEVICE_DT_GET(DT_ALIAS(accel5)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel6))
+	DEVICE_DT_GET(DT_ALIAS(accel6)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel7))
+	DEVICE_DT_GET(DT_ALIAS(accel7)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel8))
+	DEVICE_DT_GET(DT_ALIAS(accel8)),
+#endif
+#if DT_NODE_EXISTS(DT_ALIAS(accel9))
+	DEVICE_DT_GET(DT_ALIAS(accel9)),
+#endif
+};
```

Comment:
hardcoding 10 instances this way doesn't look great. First, do we have any board with 10 accelerometers? Second, why not use something like `LISTIFY` to make this more compact?

Maybe a Kconfig option could actually set the number of accelerometers to expect, to be adjusted in a board overlay if > 1.

Reply:
Rewrite with LISTIFY macro.  (I didn't know the existence of this macro. It is very helpful!)
As a result of the change, defining an alias for accelX will automatically work with this sample.
I think it's easier than defining a maximum in Kconfig because all you have to do is add an alias in the overlay.

How about this change?